ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/update/PartyRecordUpdateMapperSpec.scala

```diff
@@ -0,0 +1,290 @@
+          .value shouldBe makePartyRecordUpdate(annotationsUpdateO = Some(Replace(Map("a" -> "b"))))
+        testedMapper
+          .toUpdate(
+            prWithoutAnnotations,
+            FieldMask(Seq("party_details.metadata.annotations!replace")),
+          )
+          .value shouldBe makePartyRecordUpdate(annotationsUpdateO = Some(Replace(Map.empty)))
+        testedMapper
+          .toUpdate(prWithAnnotations, FieldMask(Seq("party_details.metadata.annotations!merge")))
+          .value shouldBe makePartyRecordUpdate(annotationsUpdateO =
+          Some(Merge.fromNonEmpty(Map("a" -> "b")))
+        )
+        testedMapper
+          .toUpdate(
+            prWithoutAnnotations,
+            FieldMask(Seq("party_details.metadata.annotations!merge")),
+          )
+          .left
+          .value shouldBe UpdatePathError.MergeUpdateModifierOnEmptyMapField(
+          "party_details.metadata.annotations!merge"
+        )
+        testedMapper
+          .toUpdate(prWithAnnotations, FieldMask(Seq("party_details.metadata.annotations")))
+          .value shouldBe makePartyRecordUpdate(annotationsUpdateO =
+          Some(Merge.fromNonEmpty(Map("a" -> "b")))
+        )
+        testedMapper
+          .toUpdate(prWithoutAnnotations, FieldMask(Seq("party_details.metadata.annotations")))
+          .value shouldBe makePartyRecordUpdate(annotationsUpdateO = Some(Replace(Map.empty)))
+      }
+
+      "when inexact path match on metadata annotations field" in {
+        val userWithAnnotations = makePartyRecord(annotations = Map("a" -> "b"))
+        val userWithoutAnnotations = makePartyRecord()
+        testedMapper
+          .toUpdate(userWithAnnotations, FieldMask(Seq("party_details!replace")))
+          .value shouldBe makePartyRecordUpdate(
+          annotationsUpdateO = Some(Replace(Map("a" -> "b")))
+        )
+        testedMapper
+          .toUpdate(userWithoutAnnotations, FieldMask(Seq("party_details!replace")))
+          .value shouldBe makePartyRecordUpdate(
+          annotationsUpdateO = Some(Replace(Map.empty))
+        )
+        testedMapper
+          .toUpdate(userWithAnnotations, FieldMask(Seq("party_details!merge")))
+          .value shouldBe makePartyRecordUpdate(
+          annotationsUpdateO = Some(Merge.fromNonEmpty(Map("a" -> "b")))
+        )
+        testedMapper
+          .toUpdate(userWithoutAnnotations, FieldMask(Seq("party_details!merge")))
+          .value shouldBe emptyUpdate
+        testedMapper
+          .toUpdate(userWithAnnotations, FieldMask(Seq("party_details")))
+          .value shouldBe makePartyRecordUpdate(
+          annotationsUpdateO = Some(Merge.fromNonEmpty(Map("a" -> "b")))
+        )
+        testedMapper
+          .toUpdate(userWithoutAnnotations, FieldMask(Seq("party_details")))
+          .value shouldBe emptyUpdate
+      }
+
+      "the longest matching path is matched" in {
+        val user = makePartyRecord(
```

Comment:
shouldn't it be called `party` or `partyRecord` instead?

Reply:
Fixed